avocado/core/nrunner.py

```diff
@@ -583,6 +586,7 @@ def __init__(self, identifier, runnable, status_uris=None, known_runners=None):
             known_runners = {}
         self.known_runners = known_runners
         self.spawn_handle = None
+        self._output_dir = "/tmp/avocado-tasks/{}/output".format(uuid4())
```

Comment:
What's the advantage you see in "cooking" our own directory location, instead of using Python's standard library support? https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
IIUC, TemporaryDirectory will try to remove the directory when the context is finished. And I understood that we would like to keep those dirs, right?

We could use `mkdtemp` that is not destructive, but this is creating when called and since that, we are sending the possible directory name during the "start" maybe this is not suitable otherwise all Task instance would have an empty dir left.

I tried to create only when `run()` is called, but having the name "a priori". But for sure this logic can be changed this logic if you think that is valid. What do you think?

Reply:
The fact that you're hoping to not collide based on the UUID before creating the directory is going to be safe on, say, more than 99.99999999999% of times, but it's *not* proven to be free of a race condition.

TBH, I'd rather deal with this issue of correctness, even if it requires extra code.

Reply:
@clebergnu Yes, I agree. I tought that the 0.00000001% collision problem would be safe, but fair enought, let's change.